source/MaterialXContrib/MaterialXNode/MaterialXData.h

```diff
@@ -0,0 +1,34 @@
+#ifndef MATERIALX_DATA_H
+#define MATERIALX_DATA_H
+
+#include <MaterialXCore/Document.h>
+#include <MaterialXGenShader/GenContext.h>
+
+#include "../OGSXMLFragmentWrapper.h"
+
+#include <maya/MString.h>
+
+#include <vector>
+
+struct MaterialXData
+{
+	MaterialXData(const std::string& materialXDocument, const std::string& elementPath);
```

Comment:
As mentioned before the document + element may come from the attributes in the case of file-load so it should not be the constructor which does all the work so we do not need to delete and recreate the data object and fix all the references to it. I guess another way is to add in an update(doc, element) method.